### PR TITLE
octopus: doc: document MDS cache configuration

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
backport tracker: https://tracker.ceph.com/issues/48111

---

backport of https://github.com/ceph/ceph/pull/37825
parent tracker: https://tracker.ceph.com/issues/48010

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

File: doc/cephfs/cache-configuration.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+The Metadata Server coordinates a distributed cache among all MDS and CephFS
+clients. The cache serves to improve metadata access latency and allow clients
+to safely (coherently) mutate metadata state (e.g. via `chmod`). The MDS issues
+**capabilities** and **directory entry leases** to indicate what state clients
+may cache and what manipulations clients may perform (e.g. writing to a file).
+
+The MDS and clients both try to enforce a cache size. The mechanism for
+specifying the MDS cache size is described below. Note that the MDS cache size
+is a not a hard limit. The MDS always allows clients to lookup new metadata
+which is loaded into the cache. This is an essential policy as its avoids
```
Comment (zdover23) [inline doc/cephfs/cache-configuration.rst:14]: s/as its avoids/as it avoids/

File: doc/cephfs/cache-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+perform other upkeep.
+
+The trim configurations control an internal **decay counter**. Anytime metadata
+is trimmed from the cache, the counter is incremented.  The threshold sets the
+maximum size of the counter while the decay rate indicates the exponential half
+life for the counter. If the MDS is continually removing items from its cache,
+it will reach a steady state of ``-ln(0.5)/rate*threshold`` items removed per
+second.
+
+The defaults are conservative and may need changed for production MDS with
```
Comment (zdover23) [inline doc/cephfs/cache-configuration.rst:80]: s/may need changed/may need to be changed/

**Review by zdover23 — DISMISSED** [review]

File: doc/cephfs/cache-configuration.rst (hunk 1 above)
Comment (Vicente-Cheng) [reply doc/cephfs/cache-configuration.rst]: fixed

File: doc/cephfs/cache-configuration.rst (hunk 2 above)
Comment (Vicente-Cheng) [reply doc/cephfs/cache-configuration.rst]: fixed, @zdover23 thanks for checking : )

**Comment by zdover23** [comment]
jenkins test api

**Comment by zdover23** [comment]
jenkins test make check

**Review by smithfarm — CHANGES_REQUESTED** [review]
We try to avoid introducing changes into stable branches that are not yet in master.

@Vicente-Cheng Please re-do the cherry-pick.

@zdover23 Please open a PR targeting master with your changes. When that PR is merged, we can cherry-pick it into this PR and remove the DNM.

**Comment by zdover23** [comment]
> We try to avoid introducing changes into stable branches that are not yet in master.
> 
> @Vicente-Cheng Please re-do the cherry-pick.
> 
> @zdover23 Please open a PR targeting master with your changes. When that PR is merged, we can cherry-pick it into this PR and remove the DNM.

This might be a dumb question whose answer has been given already, but it looks like Vincente has already made the two minor alterations that I suggested, so should I still do this?

**Comment by smithfarm** [comment]
> This might be a dumb question whose answer has been given already, but it looks like Vincente has already 
> made the two minor alterations that I suggested, so should I still do this?

I saw that he made the alterations here in this PR, which targets the `octopus` branch. But I don't see that he made the alterations in the `master` branch, which is decisive. 

If we fix things directly in stable branches, without fixing them in `master` first and then cherry-picking the fixes back into the stable branches, we will soon find ourselves in a situation where we release a new stable branch which will not contain the fixes. In other words, this is a good way to create regressions.

**Comment by zdover23** [comment]
> > This might be a dumb question whose answer has been given already, but it looks like Vincente has already
> > made the two minor alterations that I suggested, so should I still do this?
> 
> I saw that he made the alterations here in this PR, which targets the `octopus` branch. But I don't see that he made the alterations in the `master` branch, which is decisive.
> 
> If we fix things directly in stable branches, without fixing them in `master` first and then cherry-picking the fixes back into the stable branches, we will soon find ourselves in a situation where we release a new stable branch which will not contain the fixes. In other words, this is a good way to create regressions.

Roger that. Until now, I didn't have a clear mental model of this part of the development-git digestive system and of what had priority over what. Thanks for this.

**Comment by smithfarm** [comment]
> Roger that. Until now, I didn't have a clear mental model of this part of the development-git digestive system and of what had priority over what. Thanks for this.

For more details on this subject, see https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst

**Comment by Vicente-Cheng** [comment]
Hi @smithfarm,
I already re-do the cherry-pick.
I did not see the suggestions of @zdover23 on `master`.
Should this backport pr wait for the above suggestions to master?

**Comment by smithfarm** [comment]
@zdover23 Will you open a master PR to fix the doc issues you identified, or were you expecting @Vicente-Cheng to do that?

**Review by batrick — APPROVED** [review]

**Comment by yuriw** [comment]
[wip-yuri7-testing-2020-12-17-1028-octopus](https://trello.com/c/icl2Agjc/1090-wip-yuri7-testing-2020-12-17-1028-octopus)

**Pull request merged**